sdk/azidentity/chained_token_credential.go

```diff
@@ -62,36 +64,56 @@ func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.Token
 		token, err = cred.GetToken(ctx, opts)
 		var credErr credentialUnavailableError
 		if errors.As(err, &credErr) {
-			errList = append(errList, credErr)
+			errList = append(errList, newCredentialUnavailableError(extractCredentialName(cred), err.Error()))
 		} else if err != nil {
 			var authFailed AuthenticationFailedError
 			if errors.As(err, &authFailed) {
-				err = fmt.Errorf("Authentication failed:\n%s\n%s"+createChainedErrorMessage(errList), err)
+				// The message of the error created here would look as follows:
+				//
+				// 	ERROR in GetToken() call for <ChainedCredentialName>:
+				// 	Attempted credentials:
+				// 		<CredentialType>: <Error message>
+				// 		<CredentialType>: <Error message>
+				//      ...
+				//
+				err = fmt.Errorf("ERROR in GetToken() call for %s:%s\n\t%s: %s", c.chainedCredentialName, createChainedErrorMessage(errList), extractCredentialName(cred), err)
 				authErr := newAuthenticationFailedError(err, authFailed.RawResponse)
 				return nil, authErr
 			}
 			return nil, err
 		} else {
-			logGetTokenSuccess(c, opts)
+			log.Write(EventAuthentication, fmt.Sprintf("Azure Identity => GetToken() result for %s %s:", c.chainedCredentialName, extractCredentialName(cred)))
 			c.successfulCredential = cred
 			return token, nil
 		}
 	}
 
 	// if we reach this point it means that all of the credentials in the chain returned CredentialUnavailableError
-	credErr := newCredentialUnavailableError("Chained Token Credential", createChainedErrorMessage(errList))
-	// skip adding the stack trace here as it was already logged by other calls to GetToken()
-	addGetTokenFailureLogs("Chained Token Credential", credErr, false)
+	credErr := newCredentialUnavailableError(c.chainedCredentialName, createChainedErrorMessage(errList))
+
+	// The error logged here would look as follows:
+	//
+	// 	Authentication: Azure Identity => ERROR in GetToken() call for <ChainedCredentialName>:
+	// 	Attempted credentials:
+	// 		<CredentialType>: <Error message>
+	// 		<CredentialType>: <Error message>
+	// 		...
+	//
+	log.Writef(EventAuthentication, "Azure Identity => ERROR in GetToken() call for %s", credErr.Error())
+
 	return nil, credErr
 }
 
 func createChainedErrorMessage(errList []credentialUnavailableError) string {
-	msg := ""
+	msg := "\nAttempted credentials:\n"
 	for _, err := range errList {
-		msg += err.Error()
+		msg += fmt.Sprintf("\t%s\n", err.Error())
 	}
+	return msg[0 : len(msg)-1]
+}
 
-	return msg
+func extractCredentialName(credential azcore.TokenCredential) string {
+	return fmt.Sprintf("%T", credential)[len("*azidentity."):]
```

Comment:
Let’s get @chlowell feedback as well on this one 🙂

Reply:
Eh, this feels hacky but I like it better than
```
Attempted credentials:
        *azidentity.ManagedIdentityCredential: ...
        *azidentity.AzureCLICredential: ...
```
I suppose we could obviate the hacky by having credentials prepend e.g. "ManagedIdentityCredential authentication failed: " to their errors. That could be verbose or otherwise awkward in some cases though, so more investigation needed there (I'd call it out of scope for this PR).

Reply:
I prefer the hacky way at the moment.

Reply:
Is not it possible to have credentials in here that come from packages other than azidentity?

Reply:
That’s a good point! — I can check if `azidentity` is present and remove it only then 🤔 how does that sound?

Reply:
I'm really back to - is it _really_ worth stripping out azidentity? If we have credentials coming from separate spots it seems nice to have the package just to make it easy.

Reply:
I expect credentials from other packages to be very rare. For my part, I think it's worth removing `*azidentity.`; `strings.TrimPrefix()` will do that safely.

Reply:
@chlowell I like that! thank you!

Reply:
Updated here: https://github.com/Azure/azure-sdk-for-go/pull/16844/commits/cae3d8766225ec71ad22164f852e61f2be011a1c